src/main/java/org/opensearch/neuralsearch/query/HybridQueryBuilder.java

```diff
@@ -294,6 +325,26 @@ private Collection<Query> toQueries(Collection<QueryBuilder> queryBuilders, Quer
         return queries;
     }
 
+    private static void validatePaginationDepth(final int paginationDepth, final QueryShardContext queryShardContext) {
+        if (paginationDepth <= LOWER_BOUND_OF_PAGINATION_DEPTH) {
```

Comment:
i believe lower bound means it is valid value, i would recommend add + 1 to lower bound and change it to <

Reply:
Changing condition to `paginationDepth <= LOWER_BOUND_OF_PAGINATION_DEPTH`